openqaoa/problems/problem.py

```diff
@@ -135,8 +168,26 @@ def n(self, input_n):
 
         self._n = input_n
 
+    def set_metadata(self, metadata:dict={}):
+        """
+        Sets the metadata of the problem.
+
+        Parameters
+        ----------
+        metadata: dict
+            The metadata of the problem. All keys and values will be stored in the metadata dictionary.
+        """
+        #here we try if the dictionary is serializable
+        try:
+            _ = json.dumps(metadata)
+        except Exception as e:
+            raise e
```

Comment:
I think it is easier to check if the metadata is serializable by doing this, @Q-lds what do you think?